app/src/utils/queries.ts

```diff
@@ -29,17 +29,65 @@ const queryBuilder = (
   }
 `;
 
-const filesQuery = `
-  ... on Tree {
-    entries {
-      name
-      object {
-        ... on Blob {
-          text
+const filesQuery = `    
+... on Blob {
+  id
+  isBinary
+  text
+}
+... on Tree {
+  entries {
+    name
+    object {
+      ... on Blob {
+        id
+        text
+        isBinary
+      }
+      ... on Tree {
+        entries {
+          name
+          object {
+            ... on Blob {
+              id
+              text
+              isBinary
+            }
+            ... on Tree {
+              entries {
+                name
+                object {
+                  ... on Blob {
+                    id
+                    text
+                    isBinary
+                  }
+                  ... on Tree {
+                    entries {
+                      name
+                      object {
+                        ... on Blob {
+                          id
+                          isBinary
+                          text
+                        }
+                        ... on Tree {
+                          entries {
+                            name
```

Comment:
Add object with blob or remove this level.
If you receive an entry without object wouldn't it break?